app/views/layouts/_flash.html.erb

```diff
@@ -0,0 +1,5 @@
+<% if flash[:alert] || flash[:notice] %>
```

Comment:
Would `flash.any?` work?

Reply:
it's sometimes necessary to store variables and such in `flash`, so we tend to stick with checking for expected flash messages. not sure if that's applicable in this instance, but just sharing that.

Reply:
Note: Find built-in rails flash handler.

Reply:
we'll normally just iterate over expected keys and render a section based on that. example:
```erb
<% flash.slice(:alert, :notice).each do |type, msg| %>
  <div class="<%= type %>">
    <%= msg %>
  </div>
<% end %>
```

Reply:
the slice method is undefined. is there a certain module that I need to import?

Reply:
Since the flash is an object and not necessarily a hash, it cannot be sliced until turned into a hash. I tried using the to_hash method and that seemed to cause from errors. So instead, I used David's soln: 

```
<% flash.each do |type, msg| %>
  <div class="flash-<%= type %>">
    <%= msg %>
  </div>
<% end %>
```